code/drasil-database/lib/Database/Drasil/ChunkDB.hs

```diff
@@ -261,8 +261,13 @@ insertAllOutOfOrder12 strtr as bs cs ds es fs gs hs is js lcs rs =
                    fs', gs', hs', is', js']
     calt = concat altogether
 
-    -- Calculate what chunks are depended on (i.e., UID -> Dependants)
-    chDpdts = invert $ M.fromList $ map (\c -> (c ^. uid, S.toList $ chunkRefs c)) calt
+    -- Calculate what chunks are depended on (i.e., UID -> Dependants), but only
+    -- keep dependencies that correspond to chunks we are actually inserting or
+    -- that already exist in the chunk table. References/LabelledContent live in
+    -- separate tables and are handled elsewhere.
+    allowedDeps = S.fromList (M.keys (chunkTable strtr) ++ map (^. uid) calt)
+    chDpdts = invert $ M.fromList $
+      map (\c -> (c ^. uid, S.toList $ S.filter (`S.member` allowedDeps) (chunkRefs c))) calt
```

Comment:
I'm not quite sure I'm understanding this change. Does it not make it so that any chunk dependancy that is neither already in the `ChunkDB` nor in the list of chunks being added, is ignored? I think this would mean that we can insert chunks into the `ChunkDB` that have unsatisfied dependencies?